src/main/scala/coreplex/Coreplex.scala

```diff
@@ -65,21 +80,82 @@ abstract class Coreplex(implicit val p: Parameters, implicit val c: CoreplexConf
 }
 
 class DefaultCoreplex(tp: Parameters, tc: CoreplexConfig) extends Coreplex()(tp, tc) {
+  class TileWrapper(tileBuilder: (Bool, Parameters) => Tile, _clock: Clock, _reset: Bool)
+      (implicit p: Parameters) extends Module(Some(_clock), Some(_reset)) {
+
+    val wrappedTile = tileBuilder(reset, p)
+    val io = new Bundle {
+      val tile = wrappedTile.io.cloneType
+      val ext_clock = Clock(INPUT)
+      val ext_reset = Bool(INPUT)
+    }
+
+    io.tile.uncached <> wrappedTile.io.uncached.map(AsyncClientUncachedTileLinkTo(io.ext_clock, io.ext_reset, _))
+    io.tile.cached <> wrappedTile.io.cached.map(AsyncClientTileLinkTo(io.ext_clock, io.ext_reset, _))
+    wrappedTile.io.interrupts := LevelFrom(io.ext_clock, io.tile.interrupts)
```

Comment:
This makes me pretty nervous. There is a lot to consider here.

There are already Gateways for the interrupts. This sort of thing may be better handled there, like "AsynchronousLevelGateway", which does its own simple synchronization internally. Actually maybe it already is by the LevelGateway, should go look at what kinds of assumptions it is making there. Interrupts that come from a top-level pin should be synchronized by the Gateway anyway, so this seems like something better handled there.

Also, we need to consider what it means now if the CPU services an interrupt (indirecly making this 'interrupt' signal go low) then signals 'complete' to the PLIC. If the low level takes a long time to get through this synchronizer, the PLIC may see the interrupt again. Not saying these things are definitely problems, but you need to carefully consider this and convince yourself they AREN'T. I think that is hard to do in general.